src/mono/CMakeLists.txt

```diff
@@ -525,7 +525,7 @@ if(GCC)
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu11")
   else()
     # We require C99 with some GNU extensions, e.g. `linux` macro
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu11")
```

Comment:
Instead of passing the standard flag directly, can we change this to set the CMake properties that emit these versions (`CMAKE_C_STANDARD_REQUIRED`, `CMAKE_C_STANDARD`, and `CMAKE_C_EXTENSIONS`)?

Reply:
yes, CMAKE_C_EXTENSIONS is what I didn't know about, thanks!

And that means we can actually just inherit `CMAKE_C_STANDARD_REQUIRED` and `CMAKE_C_STANDARD` from `eng/native/configurecompiler.cmake`.  Nice.

Reply:
> ah nevermind we don't include that in Mono yet.

Yup, once it's done, it will cleanup some workarounds such as: https://github.com/dotnet/runtime/blob/c540d01de6a2656722cbe85b9fa3eac400510279/src/native/libs/System.IO.Compression.Native/extra_libs.cmake#L3 and cleanup duplicate code from cmake files under src/mono.